server/container/spring/src/main/java/org/apache/james/container/spring/bean/factorypostprocessor/MailboxConfigurationBeanFactoryPostProcessor.java

```diff
@@ -96,10 +109,24 @@ public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory)
             registry.registerAlias(mailboxIdDeserializer, "mailbox-id-deserializer");
             registry.registerAlias(mailboxIdFactory, "mailboxIdFactory");
 
+            removeMailboxManagersExceptRightSelectedOne(registry, mailbox);
+
         } catch (ConfigurationException e) {
             throw new FatalBeanException("Unable to config the mailboxmanager", e);
         }
 
     }
 
+    private void removeMailboxManagersExceptRightSelectedOne(BeanDefinitionRegistry registry, String selectedMailboxManager) {
+        for (String otherMailboxManager : MAILBOX_MANAGER_IDS) {
+            if (isNotSelectedMailboxManager(registry, selectedMailboxManager, otherMailboxManager)) {
+                registry.removeBeanDefinition(otherMailboxManager);
+            }
+        }
+    }
+
+    private boolean isNotSelectedMailboxManager(BeanDefinitionRegistry registry, String selectedMailboxManager, String otherMailboxManager) {
```

Comment:
it's not only "isNotSelected"...

I think name can be 'registeredAndNotSelected' can be better